atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -116,24 +110,15 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
     }
 
     @Override
-    public <T, E extends Exception> T runTaskWithLocksThrowOnConflict(
-            Iterable<HeldLocksToken> lockTokens,
-            LockAwareTransactionTask<T, E> task)
+    public <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionThrowOnConflict(
```

Comment:
I think this will break a lot of products. Instead of changing the interface, can we add one more method with the PreCommitCondition?

Reply:
Or maybe just add new method and make `runTaskWithLocksThrowOnConflict` use it.

Reply:
Oh, nvm, I misread this, you did exactly what I suggested on `AbstractLockAwareTransactionManager`.